### PR TITLE
feat(readme): Add README.md for html package

### PR DESCRIPTION
**Pull request by jbrundage**
Add html/BooleanInput widget
Add html/HTMLInput widget
Add html/NumberInput widget
Add html/SetInput widget
Add html/StringInput widget
Add html/FieldsetForm widget

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
I tested these widgets (and their sample) in a stand-alone html page. I will create another PR for the stand-alone html page as I think it provides some powerful testing capabilities.

### CODE REVIEW

**Comment by jbrundage** [comment]
fieldsetform

**Comment by jbrundage** [comment]
An HTMLInput can control its width with three different properties (listed here based on priority):
1. `minWidth` - minimum width of the input (pixels)
2. `widthPercentage` - percentage of the `FieldsetForm`'s width (or any other future html form) (optional)
3. `fixedWidth` - fixed pixel width (optional) (overwritten by `widthPercentage` if it exists)

If an HTMLInput widget has `.linebreak(true)` then a linebreak will be inserted before that input widget at the form level.

**Comment by jbrundage** [comment]
@GordonSmith This is somewhat of a work in progress because I'd like to abstract some things from the `FieldsetForm` into a new `api/IForm` widget.

**Comment by GordonSmith** [comment]
Is that checkbox abnormally large?

**Comment by jbrundage** [comment]
It is larger than usual, but that was just a spur of the moment design decision and a minor detail. What do you think about the widget inheritance?
- HTMLInput
    - StringInput
        - NumberInput
    - BooleanInput
- FieldsetForm
    - widgets: HTMLInput[]

**Comment by jbrundage** [comment]
Also, I was thinking that a new api widget (`IForm`) could extend `comms/Connection` and then several `html/xxxxForm` widgets could serve as the input owning widget that we discussed last week (with regard to refactoring the PropertyEditor2's fieldset vs table `InputGroup`). This topic is probably better suited for voice chat.

**Comment by GordonSmith** [comment]
Need to discuss this as I think anything "input" based should be in Form?

File: demos/gallery/samples/html/FieldsetForm.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+new Bar()
+    .target("target")
+    .columns(["Category","Value"])
+    .data([
+        ["A", 3],
+        ["B", 5],
+        ["C", 8]
+    ])
+    .render()
+    ;
```
Comment (GordonSmith) [inline demos/gallery/samples/html/FieldsetForm.js]: Doesn't look like a suitable sample?

Comment (jbrundage) [reply demos/gallery/samples/html/FieldsetForm.js]: Travis was throwing errors when trying to use FieldsetForm in this sample... it think it was related to Travis trying to pull sources from unpkg? I'll have to revisit this.

**Comment by jbrundage** [comment]
Closing in favor of #3112

**Pull request closed without merging**